### PR TITLE
Drop explicit check for file existence while creating GCS blobs

### PR DESCRIPTION
**Pull request by findinpath**
## Description

Defer the file existence check to GCS API when the blob gets actually created.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by findinpath** [comment]
Maintainer, pls test the PR with secrets

**Review by wendigo — APPROVED** [review]

**Comment by electrum** [comment]
/test-with-secrets sha=175d8f1ddf77f417e875022e310730681bee640e

File: lib/trino-filesystem/src/test/java/io/trino/filesystem/AbstractTestTrinoFileSystem.java [hunk 1]
```diff
@@ -510,7 +510,11 @@ void testOutputFile()
 
             if (isCreateExclusive()) {
                 // re-create without overwrite is an error
-                assertThatThrownBy(outputFile::create)
+                assertThatThrownBy(() -> {
```
Comment (electrum) [inline lib/trino-filesystem/src/test/java/io/trino/filesystem/AbstractTestTrinoFileSystem.java:513]: Let's change this to
```java
assertThatThrownBy(() -> outputFile.create().close())

Comment (findepi) [reply lib/trino-filesystem/src/test/java/io/trino/filesystem/AbstractTestTrinoFileSystem.java:513]: `assertThatThrownBy(outputFile::create)`?

Comment (electrum) [reply lib/trino-filesystem/src/test/java/io/trino/filesystem/AbstractTestTrinoFileSystem.java:513]: No, we need to ensure that it's closed. There is no guarantee that we throw on create.

**Review by findepi — APPROVED** [review]
% https://github.com/trinodb/trino/pull/21105/files#r1526885805

**Comment by findinpath** [comment]
Unrelated CI failure https://github.com/trinodb/trino/issues/18697

**Review by wendigo — APPROVED** [review]

**Pull request merged**